# Global-terrorism-EDA-project
coded and analysed by **@Aryaroop Majumder**
### Problem Statement

**Global terrorism** is described as the activities involving violence and crime with the intention to promote religious or political ideologies across the world. Global terrorism mainly consists of violence, threats and intimidation for pressurizing the government, International group or community in general.
Global Terrorism is a serious threat to the international community. Including, western individuals travelling or living abroad overseas and a substantial threat to indigenous individuals located near, or within unstable regions or areas of terrorist activity.

Globally, every region has recorded a higher average impact of terrorism in recent years than the early 2000s. The increase in the impact of terrorism was greatest felt in the _Middle East and North Africa, followed by sub-Saharan Africa._

There is considered to be a heightened threat of terrorist attack globally, especially against western interests and nationals from groups, or individuals motivated by the recent conflicts in these areas.
### What this project requires me to do

As a security/defense analyst, I will try to do the following:- 

- Exploratory data analysis
- Try to find out the hot zone of terrorism. 
- Handle the missing value 
- Do compare the various region…. visualize it and showing it on global map would be better to understand.

### What this project requires me to do

As a security/defense analyst, I will try to do the following:- 

- Exploratory data analysis
- Try to find out the hot zone of terrorism. 
- Handle the missing value 
- Do compare the various region…. visualize it and showing it on global map would be better to understand.

### Attribute Information

- **eventid** - A 12-digit Event ID system. First 8 numbers – date recorded -“yyyymmdd”. Last 4 numbers 
- **iyear** - The feild contains the year in which the incident occured
- **imonth** - This field contains the number of the month in which the incident occurred.
- **iday** - This field contains the numeric day of the month on which the incident occurred
- **extended** - 1 = "Yes" The duration of an incident extended more than 24 hours. 0 = "No" The duration of an incident extended less
- **country** - This field identifies the country code
- **country_txt** - This field identifies the country or location where the incident occurred.
- **region** - This field identifies the region code based on 12 regions
- **region_text** - This field identifies the region in which the incident occurred.
- **provstate** - Name (at the time of event) of the 1st order subnational administrative region
- **city** - Name of the city, village, or town in which the incident occurred
- **latitude** - The latitude of the city in which the event occurred.
- **longitude** - The longitude of the city in which the event occurred.
- **success** - Success of a terrorist strike
- **suicide** - 1 = "Yes" The incident was a suicide attack. 0 = "No" There is no indication that the incident was a suicide
- **attacktype1** - The general method of attack
- **attacktype1_txt** - The general method of attack and broad class of tactics used.
- **targettype1** - Type of target
- **targettype1_txt** - The general type of target/victim
- **targsubtype1_txt** - The more specific target category
- **target1** - The specific person, building, installation that was targeted and/or victimized
- **natlty1_txt** - The nationality of the target that was attacked
- **gname** - The name of the group that carried out the attack
- **nperps** - The total number of terrorists participating in the incident
- **weaptype1_txt** - General type of weapon used in the incident
- **weapsubtype1_txt** - More specific value for most of the Weapon Types



In [2]:
#importing the basic libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [3]:
#importing the dataset using pandas.read_csv command
data = pd.read_csv("./Global Terrorism Data (1).csv", encoding="ISO-8859-1")

In [4]:
data

eventid  iyear  imonth  iday approxdate  extended resolution  \
0       197000000001   1970       7     2        NaN         0        NaN   
1       197000000002   1970       0     0        NaN         0        NaN   
2       197001000001   1970       1     0        NaN         0        NaN   
3       197001000002   1970       1     0        NaN         0        NaN   
4       197001000003   1970       1     0        NaN         0        NaN   
...              ...    ...     ...   ...        ...       ...        ...   
181686  201712310022   2017      12    31        NaN         0        NaN   
181687  201712310029   2017      12    31        NaN         0        NaN   
181688  201712310030   2017      12    31        NaN         0        NaN   
181689  201712310031   2017      12    31        NaN         0        NaN   
181690  201712310032   2017      12    31        NaN         0        NaN   

        country         country_txt  region  ... addnotes  \
0            58  Dominican Republic       2  ...      NaN   
1           130              Mexico       1  ...      NaN   
2           160         Philippines       5  ...      NaN   
3            78              Greece       8  ...      NaN   
4           101               Japan       4  ...      NaN   
...         ...                 ...     ...  ...      ...   
181686      182             Somalia      11  ...      NaN   
181687      200               Syria      10  ...      NaN   
181688      160         Philippines       5  ...      NaN   
181689       92               India       6  ...      NaN   
181690      160         Philippines       5  ...      NaN   

                                                   scite1  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Somalia: Al-Shabaab Militants Attack Army Che...   
181687  "Putin's 'victory' in Syria has turned into a ...   
181688  "Maguindanao clashes trap tribe members," Phil...   
181689  "Trader escapes grenade attack in Imphal," Bus...   
181690  "Security tightened in Cotabato following IED ...   

                                                   scite2  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 2 ...   
181687  "Two Russian soldiers killed at Hmeymim base i...   
181688                                                NaN   
181689                                                NaN   
181690  "Security tightened in Cotabato City," Manila ...   

                                                   scite3  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 1 ...   
181687  "Two Russian servicemen killed in Syria mortar...   
181688                                                NaN   
181689                                                NaN   
181690                                                NaN   

                        dbsource  INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0                           PGIS        0         0        0       0      Na

In [5]:
# Information of the dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181691 entries, 0 to 181690
Columns: 135 entries, eventid to related
dtypes: float64(55), int64(22), object(58)
memory usage: 187.1+ MB


In [6]:
# Taking a peek at the data.

data.head()


eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

## Data Cleaning

The data has 135 columns which contains mostly null values over 50% and cannot be used for any data analysis.

Hence I am only using the usefull colums that i mentioned in the **readme**

In [7]:
# Selecting only the required dataset
data.rename(columns={'iyear':'Year','imonth':'Month','city':'City','iday':'Day','country_txt':'Country','region_txt':'Region','attacktype1_txt':'AttackType','target1':'Target','nkill':'Killed','nwound':'Wounded','summary':'Summary','gname':'Group','targtype1_txt':'Target_type','weaptype1_txt':'Weapon_type','motive':'Motive'},inplace=True)
data['Casualities'] = data.Killed + data.Wounded
data=data[['Year','Month','Day','Country','Region','City','latitude','longitude','AttackType','Killed','Wounded','Casualities','Target','Group','Target_type','Weapon_type']]

data.head()

Year  Month  Day             Country                       Region  \
0  1970      7    2  Dominican Republic  Central America & Caribbean   
1  1970      0    0              Mexico                North America   
2  1970      1    0         Philippines               Southeast Asia   
3  1970      1    0              Greece               Western Europe   
4  1970      1    0               Japan                    East Asia   

            City   latitude   longitude                      AttackType  \
0  Santo Domingo  18.456792  -69.951164                   Assassination   
1    Mexico city  19.371887  -99.086624     Hostage Taking (Kidnapping)   
2        Unknown  15.478598  120.599741                   Assassination   
3         Athens  37.997490   23.762728               Bombing/Explosion   
4        Fukouka  33.580412  130.396361  Facility/Infrastructure Attack   

   Killed  Wounded  Casualities                   Target  \
0     1.0      0.0          1.0             Julio Guzman   
1     0.0      0.0          0.0  Nadine Chaval, daughter   
2     1.0      0.0          1.0                 Employee   
3     NaN      NaN          NaN             U.S. Embassy   
4     NaN      NaN          NaN           U.S. Consulate   

                                Group                  Target_type Weapon_type  
0                              MANO-D  Private Citizens & Property     Unknown  
1  23rd of September Communist League      Government (Diplomatic)     Unknown  
2                             Unknown          Journalists & Media     Unknown  
3                             Unknown      Government (Diplomatic)  Explosives  
4                             Unknown      Government (Diplomatic)  Incendiary

In [8]:
# Getting the info of the null values in the dataset
data.isnull().sum()
data.isna().sum()*100/len(data) #printing out the percent of null values in the data

Year           0.000000
Month          0.000000
Day            0.000000
Country        0.000000
Region         0.000000
City           0.238867
latitude       2.507554
longitude      2.508104
AttackType     0.000000
Killed         5.676120
Wounded        8.977330
Casualities    9.287196
Target         0.350045
Group          0.000000
Target_type    0.000000
Weapon_type    0.000000
dtype: float64

## Inferential statistics

In [15]:
# Getting the basic staistics of the data that i have recieved.

# Univariate Analysis

data["Killed"].describe() 

count    171378.000000
mean          2.403272
std          11.545741
min           0.000000
25%           0.000000
50%           0.000000
75%           2.000000
max        1570.000000
Name: Killed, dtype: float64

In [16]:
data["Wounded"].describe()

count    165380.000000
mean          3.167668
std          35.949392
min           0.000000
25%           0.000000
50%           0.000000
75%           2.000000
max        8191.000000
Name: Wounded, dtype: float64

In [17]:
data["Casualities"].describe()

count    164817.000000
mean          5.296128
std          42.069023
min           0.000000
25%           0.000000
50%           1.000000
75%           4.000000
max        9574.000000
Name: Casualities, dtype: float64